lib/collection/src/update_handler.rs

```diff
@@ -503,15 +502,11 @@ impl UpdateHandler {
             };
 
             trace!("Attempting flushing");
-            let wal_flash_job = wal.lock().flush_async();
+            let wal_flash_result = wal.lock().flush();
```

Comment:
We are holding WAL lock a little bit differently (I guess, the task spawned by the `flush_open_segment_async` would have to lock WAL anyway), but other than that, I think, it should be fine.

But I'd still prefer using `tokio::task::spawn_blocking` over a blocking call.

Reply:
Aaand... I'm wrong! It does change the behavior.

`flush_async` don't hold any locks. It's implemented with a little bit of good ol' `unsafe`, but, tbh, I'm not sure which one is more correct. :/

Reply:
> flush_async don't hold any locks. It's implemented with a little bit of good ol' unsafe. But, tbh, I'm not sure the implementation is correct. :/

Yeah, I'm also having difficulty in seeing WAL as being sound. I'm currently going over the codebase, which is why I noticed this call (PR).